libkbfs/folder_block_ops.go

```diff
@@ -893,18 +948,47 @@ func (fbo *folderBlockOps) removeDirEntryInCacheLocked(lState *lockState,
 	cacheEntry.dirEntry.Ctime = now
 
 	fbo.deCache[dir.tailPointer().Ref()] = cacheEntry
+
+	// If the removed target is a directory, we can delete the cache
+	// entry for it as well, since there won't be any open handles to
+	// it.  We don't want this to show up as a dirty dir during
+	// syncing later.
+	restoreDirFn := func() {}
```

Comment:
Good call, I had no idea, was trying to avoid useless lines of code.  Thanks for writing those benchmarks!